### PR TITLE
[Signals] Fix generic methods types + better docs

### PR DESCRIPTION
**Pull request by Quelu**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://www.typescriptlang.org/docs/handbook/generics.html
  - https://millermedeiros.github.io/js-signals/docs/symbols/Signal.html

There were typescript errors because of #20429. I fixed the generic type of the methods in the Signal interface by creating a new generic type extending the class one for each function. It works but I don't understand why just passing the class generic type doesn't works. 😕 
If someone can explains this I'd be happy 😋

### CODE REVIEW

File: types/signals/index.d.ts [hunk 1]
```diff
@@ -38,47 +39,56 @@ declare namespace signals {
...
          * If Signal is active and should broadcast events.
+         * IMPORTANT: Setting this property during a dispatch will only affect the next dispatch, if you want to stop the propagation of a signal use `halt()` instead.
          */
         active: boolean;
 
         /**
          * If Signal should keep record of previously dispatched parameters and automatically
          * execute listener during add()/addOnce() if Signal was already dispatched before.
+         *
+         * @type {boolean}
```
Comment (andy-ms) [inline types/signals/index.d.ts]: This is redundant, already provided by the type declaration.

Comment (Quelu) [reply types/signals/index.d.ts]: Okay that's fixed 😉 
Do you have an answer for the question I asked in the description ?

**Comment by andy-ms** [comment]
It seems to work fine for me in `typescript@next` if I remove every `G extends T` and replace `G` with `T`. However, in the tests `myObject.started` is a `Signal<any>` so it will be compatible with `Signal<S>` any `S` (`any` is unsafe); the tests would be better if it were e.g. a `Signal<string>`.

**Comment by Quelu** [comment]
Yes there is no compilation errors when you replace `G extends T` by `T`. But when using the method `add` on a signal, the params aren't typed by what you defined on the signal generic types but by `params: {}`.
I think I understand better now why the `G extends T`. In the `add` method, you pass the generic type `T` as a parameter of the `listener` parameter so you have to pass the generic type in the context of the `add` method. But if you add just `<T>` for the method you're just creating a new empty generic type so you have to extends the class one to pass it through the method.

Don't know if i explained clearly 😋

**Comment by andy-ms** [comment]
You don't need type parameters on the method at all. Use `add(...)` not `add<T>(...)`.
Hard to tell what you're trying to accomplish without a test case that would fail without `<G extends T>`.

**Comment by RyanCavanaugh** [comment]
@Quelu Please address the merge conflict.

**Review by minestarks — CHANGES_REQUESTED** [review]
Please add at least one test case that would fail without your change, to demonstrate why the fix is needed.

**Comment by minestarks** [comment]
I'm closing this PR due to lack of activity. If you would still like to get in these changes, please open a new PR. Thank you!

**Pull request closed without merging**